src/main/java/com/pecacm/backend/services/SupportService.java

```diff
@@ -0,0 +1,69 @@
+package com.pecacm.backend.services;
+
+import com.pecacm.backend.entities.Event;
+import com.pecacm.backend.entities.Transaction;
+import com.pecacm.backend.entities.User;
+import com.pecacm.backend.enums.EventRole;
+import com.pecacm.backend.repository.TransactionRepository;
+import com.pecacm.backend.response.EventUserDetails;
+import com.pecacm.backend.response.UserEventDetails;
+import com.pecacm.backend.response.SupportUserResponse;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class SupportService {
+    private final TransactionRepository transactionRepository;
+
+    public SupportService(TransactionRepository transactionRepository) {
+        this.transactionRepository = transactionRepository;
+    }
+
+    public SupportUserResponse getSupportUserDetails(User user, PageRequest pageRequest) {
+        Page<Transaction> transactions = transactionRepository.findByUserId(user.getId(), pageRequest);
+        List<UserEventDetails> events = new ArrayList<>();
+        transactions.forEach(transaction -> {
+            Event event = transaction.getEvent();
+            UserEventDetails userEventDetails = new UserEventDetails(event.getId(), event.getTitle(), transaction.getRole(), transaction.getXp(), event.getEndDate());
+            events.add(userEventDetails);
+        });
```

Comment:
A better way of doing this is using java streams and collectors
```suggestion
        List<UserEventDetails> events = transactions.stream()
    .map(transaction -> {
        Event event = transaction.getEvent();
        return new UserEventDetails(event.getId(), event.getTitle(), transaction.getRole(), transaction.getXp(), event.getEndDate());
    })
    .collect(Collectors.toList());
```

Reply:
Replaced redundant for loops with streams